src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKReadFilterPluginDescriptor.java

```diff
@@ -182,6 +187,9 @@ public boolean isDependentArgumentAllowed(final Class<?> dependentClass) {
             ReadFilter rf = readFilters.get(s);
             if (rf != null) {
                 filters.add(rf);
+            } else if (disableToolDefaultReadFilters) {
+                // include in the read instances since the tool one is disabled, but it is provided by the user
+                filters.add(toolDefaultReadFilters.get(s));
```

Comment:
The use cases you mention are reasonable, but I think just making this change will result in duplicates in the merged filter. I think my updated changes handle this (I also added a test for the first bullet point).